core/data/src/main/java/com/nibalk/tasky/core/data/utils/BitmapUriExt.kt

```diff
@@ -5,8 +5,20 @@ import android.graphics.Bitmap
 import android.graphics.BitmapFactory
 import android.net.Uri
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.withContext
 import java.io.ByteArrayOutputStream
+import java.net.URL
+
+suspend fun URL.getCompressedByteArray(): ByteArray =
+    coroutineScope {
+        val getBytes = async(Dispatchers.IO) {
+            val url = this@getCompressedByteArray
+            url.readBytes()
+        }
+        getBytes.await()
+    }
```

Comment:
Few things:
1. The `async` doesn't do anything here if you call `await` right after. It's used to execute multiple suspending/blocking calls in parallel, but here you only have one, so it's redundant. 
2. If you don't need `async`, you also don't need `coroutineScope`. Instead, the code can look like this:
```
return withContext(Dispatchers.IO) {
    readBytes()
}
```